### PR TITLE
fix: add icon position in button

### PR DESCRIPTION
**Pull request by jainpawan21**
Issue Number: #2412 

## PR Details

1. Add iconPosition props with the right position as default value
2. Add EnumIconPosition with values Left and Right

## PR Checklist
- [ ] Tests for the changes have been added
- [ ] `npm test` doesn't throw any error

IMPORTANT: Please review the [CONTRIBUTING.md](https://github.com/amplication/amplication/blob/master/code_of_conduct.md) file for detailed contributing guidelines.

### CODE REVIEW

File: packages/amplication-design-system/src/components/Button/Button.tsx [hunk 1]
```diff
@@ -65,7 +74,7 @@ export const Button = ({
           className="amp-button__icon"
         />
       )}
-      {children}
+      {iconPosition === EnumIconPosition.Left && children}
```
Comment (yuval-hazaz) [inline packages/amplication-design-system/src/components/Button/Button.tsx:77]: if Right is the default (to keep the current behavior) then I would expect to see no change when Right is used.
I think the current options are in reverse
Isn't that the correct option here?
```suggestion
      {iconPosition === EnumIconPosition.Right && children}
```

**Review by yuval-hazaz — COMMENTED** [review]
Thanks!
please see my question in the review

File: packages/amplication-design-system/src/components/Button/Button.tsx (hunk 1 above)
Comment (jainpawan21) [reply packages/amplication-design-system/src/components/Button/Button.tsx:77]: if iconPosition === "left" then icon will should be in left and {children} should be in right direction
as line no. 77 is after icon so the icon will be rendered in the left direction and text(children) will be in right direction;

Comment (yuval-hazaz) [reply packages/amplication-design-system/src/components/Button/Button.tsx:77]: you are correct. I thought that the icon is currently on the right but it is on the left... all good!

File: package-lock.json [hunk 2]
```diff
@@ -4,7 +4,6 @@
   "requires": true,
   "packages": {
     "": {
-      "name": "amplication",
```
Comment (tupe12334) [inline package-lock.json:7]: Why did you deleted the name in the package-lock?

**Comment by jainpawan21** [comment]
@mshidlov I reverted back package-lock.json changes.
Please review again.

**Comment by mshidlov** [comment]
@jainpawan21 thank you 👍

**Review by yuval-hazaz — APPROVED** [review]

File: packages/amplication-design-system/src/components/Button/Button.tsx (hunk 1 above)
Comment (tupe12334) [inline packages/amplication-design-system/src/components/Button/Button.tsx:77]: What do you do if you don't have icon at all?

**Review by tupe12334 — APPROVED** [review]

**Pull request merged**